weldx/geometry.py

```diff
@@ -99,255 +98,402 @@ def _to_list(var) -> list:
     return [var]
 
 
-# LineSegment -----------------------------------------------------------------
+# DynamicBaseSegment ----------------------------------------------------------
 
 
-class LineSegment:
-    """Line segment."""
+class DynamicBaseSegment:
+    """Segment class to define arbitrary lines using the `~weldx.core.SpatialSeries`."""
 
-    @UREG.wraps(None, (None, _DEFAULT_LEN_UNIT), strict=True)
-    def __init__(self, points: pint.Quantity):
-        """Construct line segment.
+    def __init__(
+        self,
+        series: Union[
+            SpatialSeries, pint.Quantity, DataArray, str, MathematicalExpression
+        ],
+        max_coord: float = 1,
+        **kwargs,
+    ):
+        """Initialize a `DynamicBaseSegment`.
 
         Parameters
         ----------
-        points :
-            2x2 matrix of points. The first column is the
-            starting point and the second column the end point.
-
-        Returns
-        -------
-        LineSegment
+        series:
+            A `~weldx.core.SpatialSeries` that describes the trajectory of the shape
+            segment. Alternatively, one can pass every other object that is valid as
```

Comment:
That is cryptic/fragmented doc string. I want to know explicitly which types and contents I have to put here. It is fine to cross-ref things of course, but it should not be mandatory to read another class docstring first to understand this one.